qcelemental/tests/test_units.py

```diff
@@ -73,23 +118,77 @@ def test_unit_conversion_nist(from_unit, to_unit, expected):
         ("au_permittivity", "au_charge**2 / (au_length * au_energy)", "1.0"),
         ("au_time", "au_action / au_energy", "1.0"),
         ("au_velocity", "au_length * au_energy / au_action", "1.0"),
-        ("au_pressure", "J / m^3", "2.94210e13"),  # note precision loss due to absence in CODATA
+        ("au_pressure", "J / m^3", "2.94210e13"),
     ],
 )
-def test_unit_conversion_other(from_unit, to_unit, expected):
+def test_unit_conversion_other_2014(from_unit, to_unit, expected):
+    # Build values and quantize to the tolerance
+    constants = qcelemental.PhysicalConstantsContext("CODATA2014")
+    expected = Decimal(expected)
+    from_to_value = Decimal(constants.conversion_factor(from_unit, to_unit)).quantize(expected)
+    assert from_to_value == expected
+
+    # Build inverse
+    inv_expected = 1 / Decimal(expected)
+    to_from_value = Decimal(constants.conversion_factor(to_unit, from_unit))
 
+    # Expected to a relative tolerance as the number of digits plus one for rounding
+    # Using float comparisons as we are taking an (1 / float) inverse in the conversion code
+    rel_tol = float("10e-{}".format(len(expected.as_tuple().digits)))
+    assert pytest.approx(float(inv_expected), rel_tol) == float(to_from_value)
+
+
+@pytest.mark.parametrize(
+    "from_unit, to_unit, expected",
+    [
+        ("hartree", "kcal/mol", "627.509474063"),  # 627.50947406305
+        ("hartree", "kJ/mol", "2625.4996394798"),
+        ("bohr", "angstrom", "0.529177210903"),
+        ("bohr", "nanometer", "5.29177210903e-2"),
+        ("bohr", "cm", "5.29177210903e-9"),
+        ("bohr", "miles", "3.28815474444e-14"),
+        ("amu", "kg", "1.66053906660e-27"),
+        ("amu", "g", "1.66053906660e-24"),
+        ("Bohr", "bohr", "1.0"),
+        ("Angstrom", "Angstrom", "1.0"),
+        ("coulomb", "statC", "2997924580",),
+        ("debye", "e * Bohr", "0.39343027"),  # 0.3934302694
+        # Derived atomic units (https://en.wikipedia.org/wiki/Hartree_atomic_units)
```

Comment:
One more comment trying to simplify these tests:

```python
_atomic_unit_conversion_tests = [("au_1s...]

@parametrize("something",
[
   (test1)
]  + _atomic_unit_conversion_tests
)
```

So we do not need to duplicate the bottom piece.